betty/jinja2/__init__.py

```diff
@@ -214,6 +192,38 @@ def _init_i18n(self) -> None:
         )
         self.policies['ext.i18n.trimmed'] = True
 
+    @property
+    def context_class(self) -> type[Context]:  # type: ignore[override]
+        # Alias the environment, because `self` is bound to a different instance
+        # inside the class.
+        _environment = self
+
+        class _Context(Context):
+            def __init__(
+                self,
+                environment: Environment,
+                parent: dict[str, Any],
+                name: str | None,
+                blocks: dict[str, Callable[[Context], Iterator[str]]],
+                globals: MutableMapping[str, Any] | None = None,
+            ):
+                if 'citer' not in parent:
```

Comment:
Ensure that citations and breadcrumbs actually work, in a variety of situations (page templates, included 'mixins', etc)